src/path.cpp

```diff
@@ -181,14 +180,16 @@ maybe_t<wcstring> path_get_cdpath(const wcstring &dir, const wcstring &wd,
         cdpathsv.push_back(L".");
         for (wcstring next_path : cdpathsv) {
             if (next_path.empty()) next_path = L".";
-            if (next_path == L"." && !wd.empty()) {
+            if (next_path == L".") {
                 // next_path is just '.', and we have a working directory, so use the wd instead.
                 next_path = wd;
             }
 
-            // If next_path starts with ./ we need to replace the . with the wd.
-            if (string_prefixes_string(L"./", next_path) && !wd.empty()) {
+            // We want to return an absolute path (see issue 6220)
+            if (string_prefixes_string(L"./", next_path)) {
                 next_path = next_path.replace(0, 2, wd);
+            } else if (string_prefixes_string(L"../", next_path) || next_path == L"..") {
```

Comment:
Does this also work with `../../`?

Reply:
Looks like it fails with `../..`. I thought `normalize_path` handles that.

Reply:
actually it works fine, I was on a different branch.
This just needs to make the path absolute, the rest is resolved by `normalize_path`.